unsloth/__init__.py

```diff
@@ -123,77 +128,90 @@
 pass
 
 # Torch 2.4 has including_emulation
-major_version, minor_version = torch.cuda.get_device_capability()
-SUPPORTS_BFLOAT16 = (major_version >= 8)
-
-old_is_bf16_supported = torch.cuda.is_bf16_supported
-if "including_emulation" in str(inspect.signature(old_is_bf16_supported)):
-    def is_bf16_supported(including_emulation = False):
-        return old_is_bf16_supported(including_emulation)
-    torch.cuda.is_bf16_supported = is_bf16_supported
-else:
-    def is_bf16_supported(): return SUPPORTS_BFLOAT16
-    torch.cuda.is_bf16_supported = is_bf16_supported
-pass
+if torch.accelerate.current_accelerator().device_type == "cuda":
```

Comment:
Is there no other way to detect Intel GPUs? Since this means we must require torch 2.6

Reply:
Here's an example of what I'm doing in bnb to be more compatible:

https://github.com/bitsandbytes-foundation/bitsandbytes/blob/f3adf4f6a7381b992c4c86e7d56737ca11bee29c/tests/helpers.py#L18-L47